extensions/amp-a4a/0.1/template-validator.js

```diff
@@ -52,29 +52,31 @@ export class TemplateValidator extends Validator {
   validate(context, unvalidatedBytes, headers) {
 
     const body = utf8Decode(/** @type {!ArrayBuffer} */ (unvalidatedBytes));
+    const parsedResponseBody =
+    /** @type {./amp-ad-type-defs.AmpTemplateCreativeDef} */ (
+        tryParseJson(body));
 
     // If we're missing the relevant header, or headers altogether, we cannot
     // proceed. In this case, we return a NON_AMP response, since we cannot
     // ensure this template will be valid AMP. We will pass the body of the
     // response as the creative, and downstream renderers may attempt to render
     // it as a non-AMP creative within a cross-domain iframe.
-    if (!headers ||
+    if (!parsedResponseBody || !headers ||
         (headers.get(AMP_TEMPLATED_CREATIVE_HEADER_NAME) !== 'amp-mustache' &&
          headers.get(DEPRECATED_AMP_TEMPLATED_CREATIVE_HEADER_NAME) !==
            'amp-mustache')) {
       return Promise.resolve(
           /** @type {!./amp-ad-type-defs.ValidatorOutput} */ ({
             creativeData: {
-              creative: body,
+              crossDomainData: {
```

Comment:
I'm trying to understand why this extra layer is needed?

Reply:
Can't recall if I had a reason or not. So far seems to work fine without it, so I've removed it.